### PR TITLE
Add checks for kubectl failure to query nodes.

### PR DESCRIPTION
**Pull request by james-munson**
This addresses the issue #7102 - if we fail to get nodes from kubectl, the environment checking script is satisfied that every node in the empty list meets the criteria, and does not fail.
The change checks the return value from kubectl and also that the array of nodes reported is not empty.

### CODE REVIEW

File: scripts/environment_check.sh [hunk 1]
```diff
@@ -233,6 +233,16 @@ check_mount_propagation() {
 check_hostname_uniqueness() {
   hostnames=$(kubectl get nodes -o jsonpath='{.items[*].status.addresses[?(@.type=="Hostname")].address}')
 
+  if [ $? -ne 0 ]; then
+    error "kubectl get nodes failed - check KUBECONFIG setup"
+    exit 1
+  fi
+
+  if [[ ! ${hostnames[@]} ]]; then
+    warn "kubectl get nodes returned empty list - check KUBECONFIG setup"
```
Comment (ejweber) [inline scripts/environment_check.sh:242]: NIT: I feel like this should be an error (like the one above) since we are crashing out of the script with it.

**Review by ejweber — CHANGES_REQUESTED** [review]
I tried both new failure methods and they worked as expected.

```
eweber@laptop:~/longhorn> scripts/environment_check.sh 
[INFO]  Required dependencies 'kubectl jq mktemp sort printf' are installed.
[WARN]  kubectl get nodes returned empty list - check KUBECONFIG setup

eweber@laptop:~/longhorn> export KUBECONFIG=broke
eweber@laptop:~/longhorn> scripts/environment_check.sh 
[INFO]  Required dependencies 'kubectl jq mktemp sort printf' are installed.
W1122 14:06:05.513260   26649 loader.go:222] Config not found: broke
E1122 14:06:05.513903   26649 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E1122 14:06:05.514054   26649 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E1122 14:06:05.515413   26649 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E1122 14:06:05.516563   26649 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E1122 14:06:05.517915   26649 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?
[ERROR] kubectl get nodes failed - check KUBECONFIG setup
```

One NIT to answer or resolve.

File: scripts/environment_check.sh (hunk 1 above)
Comment (james-munson) [reply scripts/environment_check.sh]: Okay, done.

**Review by ejweber — APPROVED** [review]
LGTM

**Review by PhanLe1010 — APPROVED** [review]
LGTM.

Could you rebase and repush? @james-munson 

Should we also backport it to older versions?

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
We need to backport this to 1.5.x and 1.4.x if needed. @james-munson I just added backport labels to the source ticket, so please fix them as well. Thanks.

**Comment by james-munson** [comment]
@mergify backport v1.4.x v1.5.x

**Pull request merged**